lib/tasks/audit_code.rake

```diff
@@ -18,19 +18,43 @@ def rake_cmd
   ENV['BUNDLE_BIN_PATH'] ? 'bundle exec rake' : 'rake'
 end
 
+def load_file_safety
+  safety_cfg = File.exist?(SAFETY_FILE) ? YAML.load_file(SAFETY_FILE) : {}
+  file_safety = safety_cfg['file safety']
+  if file_safety.nil?
+    puts "Creating new 'file safety' block in #{SAFETY_FILE}"
+    file_safety = {}
+  end
+  file_safety
+end
+
+def update_safety_file(file_safety)
+  File.open(SAFETY_FILE, 'w') do |file|
+    # Consistent file diffs, as ruby preserves Hash insertion order since v1.9
+    list = {}
+    list['file safety'] = Hash[file_safety.sort]
+    YAML.dump(list, file) # Save changes before checking latest revisions
+  end
+end
+
+def add_new_file_to_file_safety(file_safety, f)
+  return if file_safety.key?(f)
+  file_safety[f] = {
+    'comments' => nil,
+    'reviewed_by' => nil,
+    'safe_revision' => nil
+  }
+end
+
 # Parameter max_print is number of entries to print before truncating output
 # (negative value => print all)
-def audit_code_safety(max_print = 20, ignore_new = false, show_diffs = false, show_in_priority = false, user_name = 'usr')
+def audit_code_safety(max_print = 20, ignore_new = false, show_diffs = false, show_in_priority = false, usr = 'usr', interactive = false)
   puts 'Running source code safety audit script.'
```

Comment:
If you were feeling really fancy, we could make this run in the terminal's altscreen if `interactive` is true.

See: https://stackoverflow.com/a/11024208